feg/gateway/services/n7_n40_proxy/n7/model_conversion.go

```diff
@@ -136,6 +146,52 @@ func GetSmPolicyDeleteReqBody(
 	}
 }
 
+func GetSmPolicyUpdateRequestsN7(
+	updates []*protos.UsageMonitoringUpdateRequest,
+) []*SmPolicyUpdateReqCtx {
+	updatesPerSession := make(map[string][]*protos.UsageMonitoringUpdateRequest)
+	// Sort updates per session
+	for _, update := range updates {
+		updatesPerSession[update.SessionId] = append(updatesPerSession[update.SessionId], update)
+	}
+
+	// Merge the updates for the same session into a single request
+	reqCtxs := []*SmPolicyUpdateReqCtx{}
+	for _, updatesList := range updatesPerSession {
+		firstUpdate := updatesList[0]
+		smPolicyId, err := GetSmPolicyId(firstUpdate.TgppCtx)
+		if err != nil {
+			// No 3gpp context. Cannot get the PolicyId
+			glog.Errorf("SmPolicyUpdate: Session %s does not have SmPolicyId", firstUpdate.SessionId)
+			continue
+		}
+
+		reqBody := &n7_sbi.PostSmPoliciesSmPolicyIdUpdateJSONRequestBody{
+			Ipv4Address: getSbiIpv4(firstUpdate.UeIpv4),
+			RatType:     getSbiRatType(firstUpdate.RatType),
+			AccessType:  getSbiAccessType(firstUpdate.RatType),
+		}
```

Comment:
move this block right before `reqCtxs = append(reqCtxs, &SmPolicyUpdateReqCtx{` and build the  `PostSmPoliciesSmPolicyIdUpdateJSONRequestBody` there. It is easier to read if you place how you build an object on the same place, if possible